pandas/core/common.py

```diff
@@ -1596,7 +1596,7 @@ def is_timedelta64_dtype(arr_or_dtype):
 
 
 def is_timedelta64_ns_dtype(arr_or_dtype):
-    tipo = _get_dtype_type(arr_or_dtype)
+    tipo = _get_dtype(arr_or_dtype)
```

Comment:
If `Y` is an object returned by `_get_dtype(X)` then `_get_dtype_type(X)` returns `Y.type` if I understand correctly. And as `_get_dtype_type` never returns `_TD_DTYPE`, `is_timedelta64_ns_dtype` will always return `False`.

```python
>>> pd.core.common._TD_DTYPE
Out[33]: dtype('<m8[ns]')
>>> pd.core.common._get_dtype_type(pd.core.common._TD_DTYPE)
Out[34]: numpy.timedelta64
>>> pd.core.common._get_dtype(pd.core.common._TD_DTYPE)
Out[35]: dtype('<m8[ns]')
>>> pd.core.common._TD_DTYPE == np.timedelta64
Out[36]: False
>>> pd.core.common.is_timedelta64_dtype('timedelta64[ns]')
Out[37]: True
>>> pd.core.common.is_timedelta64_ns_dtype('timedelta64[ns]')
Out[38]: False
```

Compare it with the definitions of `is_datetime64...`:

```python
def is_datetime64_dtype(arr_or_dtype):
    try:
        tipo = _get_dtype_type(arr_or_dtype)
    except TypeError:
        return False
    return issubclass(tipo, np.datetime64)

def is_datetime64_ns_dtype(arr_or_dtype):
    try:
        tipo = _get_dtype(arr_or_dtype)
    except TypeError:
        return False
    return tipo == _NS_DTYPE

def is_timedelta64_dtype(arr_or_dtype):
    tipo = _get_dtype_type(arr_or_dtype)
    return issubclass(tipo, np.timedelta64)

def is_timedelta64_ns_dtype(arr_or_dtype):
    tipo = _get_dtype_type(arr_or_dtype)
    return tipo == _TD_DTYPE
```